main.nf

```diff
@@ -84,170 +88,138 @@ if (params.help){
-
-    Channel
-        .fromPath(bwa_dir, checkIfExists: true)
-        .ifEmpty { exit 1, "BWA index directory not found: ${bwa_dir}" }
-        .set { bwa_index }
-} else if ( params.fasta ){
-    lastPath = params.fasta.lastIndexOf(File.separator)
-    bwa_base = params.fasta.substring(lastPath+1)
-
-    fasta = Channel
-        .fromPath(params.fasta, checkIfExists: true)
-        .ifEmpty { exit 1, "Fasta file not found: ${params.fasta}" }
-} else {
-    exit 1, "No reference genome specified!"
-}
-
-if( params.gtf ){
-    gtf = Channel
-        .fromPath(params.gtf, checkIfExists: true)
-        .ifEmpty { exit 1, "GTF annotation file not found: ${params.gtf}" }
-}
-
-if( params.bed ){
-    bed = Channel
-        .fromPath(params.bed, checkIfExists: true)
-        .ifEmpty { exit 1, "BED file not found: ${params.bed}" }
-}
+////////////////////////////////////////////////////
+/* --         DEFAULT PARAMETER VALUES         -- */
+////////////////////////////////////////////////////
 
-if( params.blacklist_filtering ){
-    blacklist = Channel
-        .fromPath(params.blacklist, checkIfExists: true)
-        .ifEmpty { exit 1, "Blacklist annotation file not found: ${params.blacklist}" }
-}
+// Configurable variables
+params.fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+params.bwa_index = params.genome ? params.genomes[ params.genome ].bwa ?: false : false
+params.gtf = params.genome ? params.genomes[ params.genome ].gtf ?: false : false
+params.gene_bed = params.genome ? params.genomes[ params.genome ].gene_bed ?: false : false
+params.macs_gsize = params.genome ? params.genomes[ params.genome ].macs_gsize ?: false : false
+params.blacklist = params.genome ? params.genomes[ params.genome ].blacklist ?: false : false
 
 // Has the run name been specified by the user?
 //  this has the bonus effect of catching both -name and --name
 custom_runName = params.name
-if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
+if (!(workflow.runName ==~ /[a-z]+_[a-z]+/)){
   custom_runName = workflow.runName
 }
 
-if(params.readPaths){
-    if(params.singleEnd){
-        Channel
-            .from(params.readPaths)
-            .map { row -> [ row[0], [file(row[1][0])]] }
-            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied" }
-            .into { raw_reads_configvali; raw_reads_fastqc; raw_reads_trimgalore }
-    } else {
-        Channel
-            .from(params.readPaths)
-            .map { row -> [ row[0], [file(row[1][0]), file(row[1][1])]] }
-            .ifEmpty { exit 1, "params.readPaths was empty - no input files supplied" }
-            .into { raw_reads_configvali; raw_reads_fastqc; raw_reads_trimgalore }
-    }
+////////////////////////////////////////////////////
+/* --          CONFIG FILES                    -- */
+////////////////////////////////////////////////////
+
+// Pipeline config
+ch_output_docs = file("$baseDir/docs/output.md", checkIfExists: true)
```

Comment:
We should have this in the template from now on too!

Reply:
To be honest, Im still confused as to what we should be using to stage files. See:
https://github.com/nf-core/rnaseq/pull/201#issuecomment-487968347

Reply:
I think it's important to use the value channel instead of a queue channel. That will work even in cloud settings if I understood Paolo correctly.

Reply:
Maybe we should test this a bit more in the cloud when I have merged this PR in `dev`. I haven't got an AWS account but is it something you could fire off with `-profile test`?

@ewels did you want to have a look at this PR? Or shall I merge? Would be good to have a working pipeline that can be pulled from `dev`, and tested a bit more while I polish things up.

Reply:
I could certainly try some things out yes 👍